src/main/java/network/brightspots/rcv/ResultsWriter.java

```diff
@@ -557,76 +556,68 @@ void generateOverallSummaryFiles(
+          csvPrinter.print(castVoteRecord.getPrecinct());
         }
-        Path outputPath =
-            Paths.get(
-                getOutputFilePath(
-                    csvOutputFolder,
-                    "dominion_conversion_contest",
-                    timestampString,
-                    contest.getId())
-                    + ".csv");
-        Logger.info("Writing cast vote records in generic format to file: %s...", outputPath);
-        CSVPrinter csvPrinter;
-        BufferedWriter writer = Files.newBufferedWriter(outputPath);
-        csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT);
-        // print header:
-        // ContestId, TabulatorId, BatchId, RecordId, Precinct, Precinct Portion, rank 1 selection,
-        // rank 2 selection, ... rank maxRanks selection
-        csvPrinter.print("Contest Id");
-        csvPrinter.print("Tabulator Id");
-        csvPrinter.print("Batch Id");
-        csvPrinter.print("Record Id");
-        csvPrinter.print("Precinct");
-        csvPrinter.print("Precinct Portion");
-        Integer numRanks = contest.getMaxRanks();
-        for (int rank = 1; rank <= numRanks; rank++) {
-          String label = String.format("Rank %d", rank);
-          csvPrinter.print(label);
+        if (castVoteRecord.getPrecinctPortion() == null) {
+          csvPrinter.print("");
+        } else {
+          csvPrinter.print(castVoteRecord.getPrecinctPortion());
         }
+        printRankings(undeclaredWriteInLabel, numRanks, csvPrinter, castVoteRecord);
         csvPrinter.println();
-        // print rows:
-        for (CastVoteRecord castVoteRecord : castVoteRecords) {
-          csvPrinter.print(castVoteRecord.getContestId());
-          csvPrinter.print(castVoteRecord.getTabulatorId());
-          csvPrinter.print(castVoteRecord.getBatchId());
-          csvPrinter.print(castVoteRecord.getId());
-          if (castVoteRecord.getPrecinct() == null) {
-            csvPrinter.print("");
-          } else {
-            csvPrinter.print(castVoteRecord.getPrecinct());
-          }
-          if (castVoteRecord.getPrecinctPortion() == null) {
-            csvPrinter.print("");
-          } else {
-            csvPrinter.print(castVoteRecord.getPrecinctPortion());
-          }
-          printRankings(undeclaredWriteInLabel, contest, csvPrinter, castVoteRecord);
-          csvPrinter.println();
-        }
-        // finalize the file
-        csvPrinter.flush();
-        csvPrinter.close();
-        filesWritten.add(outputPath.toString());
-        Logger.info("Successfully wrote: %s", outputPath.toString());
-
-        File file = new File(outputPath.toString());
-        boolean readOnlySucceeded = file.setReadOnly();
-        if (!readOnlySucceeded) {
-          Logger.warning("Failed to set file to read-only: %s", file.getAbsolutePath());
-        }
+      }
+      // finalize the file
+      csvPrinter.flush();
+      csvPrinter.close();
+      filesWritten.add(outputPath.toString());
```

Comment:
Remind me why we're using a List here if it only ever contains a single value...?

Reply:
No idea! Looks like the previous code was a loop that looked like it could have multiple values, but filtered for just one, but I'll just replace it with a String rather than going down the gitlog rabbithole.